src/Features/Core/Portable/IntroduceVariable/AbstractIntroduceParameterDocumentRewriter.cs

```diff
@@ -512,6 +512,10 @@ private async Task<SyntaxNode> GenerateNewMethodOverloadAsync(int insertionIndex
                         continue;
 
                     var invocationArguments = _syntaxFacts.GetArgumentsOfArgumentList(argumentListSyntax);
+
+                    if (insertionIndex > invocationArguments.Count)
```

Comment:
crash fix

Reply:
This is a very narrow case, but it seems to me, it doesn't make sense to try to introduce new parameter based on an argument to a recursive call to the containing method itself. I wonder if we need a check upfront (e.g. in ComputeRefactoringsAsync)

Reply:
That was a question I had as well but figured if someone tried it, then I'm sure other people would try it as well.